### ISSUE TITLE
Output plugin either doesn't work at all or trigger incorrect output after the 32th plugin

### ISSUE THREAD

**Issue by wtchangdm** [fluent/fluent-bit#2898]
## Bug Report

**Describe the bug**
We currently send logs to different CloudWatch Log groups based on the tag.
However, as our destinations grow, we need to send more than 32 log groups.

We noticed that after the 32th "Output" block, fluent-bit starts acting weird.
More precisely, the 33th output plugin won't work at all. And if you add the 34th output, it actually triggers the first output plugin no matter it matches the tag or not.

After some investigations, I found out that is not plugin-specific. It's possible to be reproduced with simple configurations.

Thanks in advance, and happy new year.

**To Reproduce**
- Rubular link if applicable: N/A
- Example log message if applicable:
```
[0] dummy: [1609409366.867601200, {"message"=>"dummy"}]
```
- Steps to reproduce the problem:

1. Create `fluent-bit.conf` with config below.
2. create a empty file `log.txt`
3. Start fluent-bit with command: `$ docker run --rm -p 7777:7777 -v $PWD/fluent-bit.conf:/fluent-bit/etc/fluent-bit.conf:ro -v $PWD/log.txt:/test/log/log.txt fluent/fluent-bit:1.6.9`
4. Observe the stdout, the 33th stdout plugin shows nothing.
5. Add another null output on the above of stdout plugin, and restart the container.
6. Observe the stdout, it actually prints dummy logs. However, when you check on log.txt, it also shows these logs **but the tag doesn't even match**.

**Expected behavior**

Output plugin should work beyond 32 of them.

**Screenshots**

N/A

**Your Environment**

* Version used: 1.6.3 and 1.6.9
* Configuration: Please see the full config below.
* Environment name and version (e.g. Kubernetes? What version?):
* Server type and version: k8s (eks v1.16) and docker desktop on Mac (engine: 20.10.0)
* Operating System and version: Amazon Linux 2 (for EKS) and macOS 11.1 w/ Docker version mentioned above.
* Filters and plugins:

```
[SERVICE]
    flush        1

[INPUT]
    Name        dummy
    Listen      0.0.0.0
    Tag         dummy
#0
[OUTPUT]
    Name  File
    Path  /test/log/
    File  log.txt
    Match not_dummy
#1
[OUTPUT]
    Name null
    Match not_dummy
#2
[OUTPUT]
    Name null
    Match not_dummy
#3
[OUTPUT]
    Name null
    Match not_dummy
#4
[OUTPUT]
    Name null
    Match not_dummy
#5
[OUTPUT]
    Name null
    Match not_dummy
#6
[OUTPUT]
    Name null
    Match not_dummy
#7
[OUTPUT]
    Name null
    Match not_dummy
#8
[OUTPUT]
    Name null
    Match not_dummy
#9
[OUTPUT]
    Name null
    Match not_dummy
#10
[OUTPUT]
    Name null
    Match not_dummy
#11
[OUTPUT]
    Name null
    Match not_dummy
#12
[OUTPUT]
    Name null
    Match not_dummy
#13
[OUTPUT]
    Name null
    Match not_dummy
#14
[OUTPUT]
    Name null
    Match not_dummy
#15
[OUTPUT]
    Name null
    Match not_dummy
#16
[OUTPUT]
    Name null
    Match not_dummy
#17
[OUTPUT]
    Name null
    Match not_dummy
#18
[OUTPUT]
    Name null
    Match not_dummy
#19
[OUTPUT]
    Name null
    Match not_dummy
#20
[OUTPUT]
    Name null
    Match not_dummy
#21
[OUTPUT]
    Name null
    Match not_dummy
#22
[OUTPUT]
    Name null
    Match not_dummy
#23
[OUTPUT]
    Name null
    Match not_dummy
#24
[OUTPUT]
    Name null
    Match not_dummy
#25
[OUTPUT]
    Name null
    Match not_dummy
#26
[OUTPUT]
    Name null
    Match not_dummy
#27
[OUTPUT]
    Name null
    Match not_dummy
#28
[OUTPUT]
    Name null
    Match not_dummy
#29
[OUTPUT]
    Name null
    Match not_dummy
#30
[OUTPUT]
    Name null
    Match not_dummy
#31
[OUTPUT]
    Name null
    Match not_dummy

#32
# The 33th output, won't work at all
# If you put another null output above, this will actually trigger the first output plugin,
# no matter it matches the tag or not.
[OUTPUT]
    Name stdout
    Match dummy
```

**Additional context**

We actually use fluent-bit to send logs by namespaces for a k8s cluster. As the namespaces increase, we end up using a lot of outputs using Helm Chart like:

```
[OUTPUT]
    Name cloudwatch_logs
    Match kube__test-namespace__*
    region ap-northeast-1
    auto_create_group On
    log_group_name /some-log-group/test-namespace
    log_stream_name test-namespace-via-${HOSTNAME}
```

**Comment by wtchangdm** [comment]
While this pattern is not very common, I think it's still worth a check?

**Comment by nokute78** [comment]
Could you check #2914 ?
The patch #2882 is released from v1.7.0.

**Comment by wtchangdm** [comment]
@nokute78 I've tried with 1.7.2, it works. Thanks for the head's up.
